cypress/integration/networking.spec.ts

```diff
@@ -3,49 +3,165 @@ import * as path from "path";
+
+            commands.pick("sidebar", "search").click();
+            commands.pick("search-input").type(urn || "");
+            commands.pick("follow-toggle").click();
+
+            cy.log("project is briefly shown in the waitingroom");
+            commands
+              .pick("following-tab")
+              .contains("Still looking")
+              .should("exist");
+            commands
+              .pick("following-tab")
+              .contains(urn.replace("rad:git:", ""))
+              .should("exist");
+
+            // TODO: check that notifications are shown
+          });
+
+          cy.log("project moved out of the waiting area and is available");
           commands
             .pick("following-tab", "project-list-entry-new-fancy-project.xyz")
-            .should("exist");
-        });
-      });
+            .click();
+
+          cy.log("the maintainer shows up in the peer selector");
+          commands.pick("peer-selector").contains("rudolfs").should("exist");
+          commands.pick("peer-selector").contains("maintainer").should("exist");
+          commands.pick("peer-selector").click();
+
+          cy.log("the current user does not show up in the peer selector");
+          commands
+            .pick("peer-dropdown-container")
+            .contains("abbey")
+            .should("not.exist");
+
+          cy.log("add contributor remote on maintainer's node");
+          commands.pick("sidebar", "profile").click();
+          commands.pick("peer-id").then(el => {
+            const peerId = el.attr("title");
+            if (!peerId) {
+              throw new Error("Could not find PeerId");
+            }
+            const shortenedPeerId = peerId.slice(0, 7);
+
+            nodeManager.asNode(maintainerNode);
+            commands.pick("project-list-entry-new-fancy-project.xyz").click();
+
+            commands.pick("peer-selector").click();
+            commands.pick("manage-remotes").click();
+
+            commands.pick("followed-peers", "peer-abbey").should("not.exist");
+            cy.contains("Still looking").should("not.exist");
+
+            commands.pick("peer-input").type(peerId);
+            commands.pick("follow-button").click();
+
+            cy.log("the remote shows up in the waiting area");
+            cy.contains("Still looking").should("exist");
+            commands
+              .pick("pending-peers")
+              .contains(shortenedPeerId)
+              .should("exist");
+
+            cy.log("the remote moves out of the waiting area");
+            commands
+              .pick("followed-peers", "peer-abbey")
+              .contains(shortenedPeerId)
+              .should("exist");
+            cy.contains("Still looking").should("not.exist");
+            commands
+              .pick("followed-peers", "peer-abbey", "follow-toggle")
+              .contains("Following")
+              .should("exist");
+          });
+
+          cy.log("add a random non-existing remote");
```

Comment:
This should probably be a separate test case.

Reply:
I'm not sure yet how to effectively split a large integration spec up into multiple smaller ones without sacrificing performance because all steps depend on all of the previous steps.

https://docs.cypress.io/guides/references/best-practices.html#Creating-%E2%80%9Ctiny%E2%80%9D-tests-with-a-single-assertion

Reply:
This is indeed correct for interdependent steps. But in our case the last part only depends on project creation and not on the rest of the test and doesn’t even need a second node. Splitting it into a separate test case will make it easier to grasp the test suite and it will be easier to isolate failures and fix them. The only disadvantage would be that we have to repeat the project setup. But I guess for the last part using `commands.createProjectWithFixture()` is sufficient.

All in all, I don’t think this is a major issue though, so feel free to keep it as is if your not convinced.

Reply:
Ha, I just realised that we have all of this already covered by: https://github.com/radicle-dev/radicle-upstream/blob/master/cypress/integration/project/peer_management.spec.ts#L24. I was just mechanically translating the QA steps into this spec and didn't look around properly.

Thanks for being persistant!

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/1559/commits/3f28102219c1cc1be7391a57043be9627e3b7fb0